### PR TITLE
Adding the Places Pipeline.

### PR DESCRIPTION
**Pull request by zem360**
# Tell us what you do here

- [x] contributing new pipeline: Places Pipeline

# Relevant issue

issue #130

### CODE REVIEW

File: pipelines/placesapi/__init__.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    '''
+    The Parameters to be passsed to endpoint:
+    @:params: query - The text against which a search will be made.
+    @:params: radius  - Defines the distance (in meters) within which to return place results.
+    @:params: api_secret_key - The api key for the Places API defined in dlt.secrets.
+    '''
+    payload = {
+        'query' : 'FitX',
+        'radius' : '',
```
Comment (burnash) [inline pipelines/placesapi/__init__.py:41]: Could you elaborate on the purpose of the empty string here? Does API expect this?

File: pipelines/placesapi/__init__.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        'key': {api_secret_key}
```
Comment (burnash) [inline pipelines/placesapi/__init__.py:42]: The 'key' key in the payload dictionary is wrapped in curly braces which would make it a set in Python. If the API expects a string, remove the curly braces.

File: pipelines/placesapi/__init__.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+
+@dlt.resource(write_disposition="append")
+def nearby_search(api_secret_key=dlt.secrets.value):
+    """
+    A Nearby Search lets you search for places within a specified area. You can refine your search request by supplying keywords or specifying the type of place you are searching for.
+    For more Info: "https://developers.google.com/maps/documentation/places/web-service/search-nearby"
+    
+    """
+    pass
```
Comment (burnash) [inline pipelines/placesapi/__init__.py:69]: What is the purpose of these two placeholder functions?

**Review by burnash — CHANGES_REQUESTED** [review]
Thanks for the contribution! I have a couple of little suggestions to make the code even better, though. But overall, you're doing great!

File: pipelines/placesapi/README.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Google Places
+***
+
+The pipleline uses Googlle Places API to search information about different locations.
```
[the remark this answers is not in the document]
Comment (zem360) [reply pipelines/placesapi/README.md]: Fixed the Typo.

File: pipelines/placesapi/__init__.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+    # API Endpoint.
+    text_search_url = "https://maps.googleapis.com/maps/api/place/textsearch/json?"
+
+    '''
+    The Parameters to be passsed to endpoint:
+    @:params: query - The text against which a search will be made.
+    @:params: radius  - Defines the distance (in meters) within which to return place results.
+    @:params: api_secret_key - The api key for the Places API defined in dlt.secrets.
+    '''
```
[the remark this answers is not in the document]
Comment (zem360) [reply pipelines/placesapi/__init__.py]: Fixed this. Now part of the docstring.

File: pipelines/placesapi/__init__.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    payload = {
+        'query' : 'FitX',
```
[the remark this answers is not in the document]
Comment (zem360) [reply pipelines/placesapi/__init__.py]: Adding Place holder query like : "Add search text here".

File: pipelines/placesapi/__init__.py (hunk 1 above)
Comment (zem360) [reply pipelines/placesapi/__init__.py]: These are optional parameters. I have added more info in the docstring for this now.

File: pipelines/placesapi/__init__.py (hunk 2 above)
Comment (zem360) [reply pipelines/placesapi/__init__.py]: Fixed.

File: pipelines/placesapi/__init__.py (hunk 3 above)
Comment (zem360) [reply pipelines/placesapi/__init__.py]: I added this for future implementation as placeholders, but have removed. them in the new commit.

File: pipelines/placesapi/README.md
[the remark this answers is not in the document]
Comment (zem360) [reply pipelines/placesapi/README.md:1]: Added more detail to README file.

File: pipelines/placesapi/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+Running this command will create a directory with the following structure:
+
+```bash
+├── .dlt
+│   ├── .pipelines
+│   ├── config.toml
+│   └── secrets.toml
+├── placesapi
+│   └── __pycache__
```
Comment (burnash) [inline pipelines/placesapi/README.md:28]: When everything is set up correctly `__pycache__` should not be here. Let me know if it's not ignored by gitignore in your case.

File: pipelines/placesapi/__init__.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    @:params: radius (optional)  - Defines the distance (in meters) within which to return place results. You may bias results to a specified circle by passing a location and a radius parameter. Default value is 50,000 meters for text search.
+    @:params: location (optional) - Can be used with the radius parameter to bias result.
+    @:params: api_secret_key - The api key for the Places API defined in dlt.secrets.
+    """
+
+    # API Endpoint.
+    text_search_url = "https://maps.googleapis.com/maps/api/place/textsearch/json?"
+
+    payload = {
+        'query' : 'ADD SEARCH TEXT HERE',
```
Comment (burnash) [inline pipelines/placesapi/__init__.py:38]: We can improve the usability of this resource if we let the user of the pipeline pass these parameters via the `text_search()` arguments. Please take a look at how resources are implemented in the Chess pipeline: https://github.com/dlt-hub/pipelines/blob/beeed5001656fd42c7da7f0b955e6fd8be43ebae/pipelines/chess/__init__.py#L49-L50

**Review by burnash — CHANGES_REQUESTED** [review]
Great job so far! There are a few more changes needed, as detailed in the comments. Your work is pushing us in the right direction. Let's keep refining and finish this up!

**Review by burnash — CHANGES_REQUESTED** [review]
Great work on this pull request! We're almost there. There are just a few minor areas to fine-tune for best practices. Please review the comments I left in the code. Thanks for your efforts!

File: pipelines/placesapi_pipeline.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    # pretty print the information on data that was loaded
+    print(load_text_search)
+
+
+if __name__=='__main__':
+
+    # Add Search Parameters 
+    search_query = 'Add Search Text Here!'
+    search_radius =  'Add Search Radius Here!'
+    search_location = ('Latitude, Longitude')
```
Comment (burnash) [inline pipelines/placesapi_pipeline.py:28]: In case `search_location` is a string, parentheses are redundant.

**Review by burnash — CHANGES_REQUESTED** [review]
Great work on improvements! There's one small improvement before it's ready for merging.

File: pipelines/placesapi_pipeline.py (hunk 9 above)
Comment (burnash) [reply pipelines/placesapi_pipeline.py:28]: Also as this is a demo script, please consider using a real coordinates, search query and radius so it works right after a user fetches the pipeline with `dlt init`. See Chess pipeline demo an example: https://github.com/dlt-hub/pipelines/blob/master/pipelines/chess_pipeline.py

**Pull request closed without merging**